src/microbe_stage/ColonyCompoundBag.cs

```diff
@@ -26,34 +28,38 @@ public float GetCapacityForCompound(Compound compound)
     public void DistributeCompoundSurplus()
     {
         var bags = GetCompoundBags().ToList();
-        using var compounds = GetEnumerator();
```

Comment:
The |GetEnumerator| function has been overwritten at line 75, will it still return the same value as |foreach x in this| ?

Reply:
AFAIK using foreach is the equivalent to:
```
var enumerator = value.GetEnumerator();
while(enumerator.Next())
{
   var item = enumerator.Current;
    ...
}
enumerator.Dispose();
```

With this being the foreach statement:
```
foreach(var item in value)
{
    ...
}
```